modules/api/src/membership/controllers/membership.controller.ts

```diff
@@ -0,0 +1,27 @@
+import {Controller, UseGuards, Get, Post, Body} from '@nestjs/common';
+import { JwtAuthGuard } from '../../auth/guards/jwt-auth.guard';
+import {MembershipService} from '../services/membership.service';
+import {CreateMembershipDto} from '../../dto/membership/create-membership.dto';
+
+@Controller('membership')
+export class MembershipController {
+  constructor(
+      private readonly membershipService: MembershipService,
+  ) {}
+
+  @UseGuards(JwtAuthGuard)
+  @Post()
+  async createMembership(
+      @Body() {userId, kitchenId, membership}: CreateMembershipDto,
+  ) {
+    return await this.membershipService.saveNew(userId, kitchenId, membership);
```

Comment:
this method should just take the CreateMembershipDTO instead of the destructured objects. better to pass 1 object than 3 other objects.

Reply:
sure we can do that. I thought it would be clearer to break down the peices.